contracts/Crowdsale/BasicCrowdsale.sol

```diff
@@ -170,16 +121,34 @@ contract BasicCrowdsale is MintedCrowdsale, FinalizableCrowdsale {
         require(beneficiaries.length == amount.length);
     }
 
-    function extendPrivateSaleDuration(uint256 extentionInSeconds) public onlyOwner returns (bool) {
+    /**
+        @param extentionInDays is a simple number of the days, e.c. 3 => 3 days
+     */
+    function extendPrivateSaleDuration(uint256 extentionInDays) public onlyOwner returns (bool) {
         require(now <= privateSaleEndDate);
-        privateSaleEndDate = privateSaleEndDate.add(extentionInSeconds);
-        closingTime = closingTime.add(extentionInSeconds);
-        emit LogPrivateSaleExtended(extentionInSeconds);
+        extentionInDays = extentionInDays.mul(1 days); // convert the days in minutes
+        privateSaleEndDate = privateSaleEndDate.add(extentionInDays);
+        closingTime = closingTime.add(extentionInDays);
+        emit LogPrivateSaleExtended(extentionInDays);
+        return true;
+    }
+
+    /**
+        @param extentionInDays is a simple number of the days, e.c. 3 => 3 days
+     */
+    function extendMainSailDuration(uint256 extentionInDays) public onlyOwner returns (bool) {
+        require(now > privateSaleEndDate);
+        require(!hasClosed());
+        require(mainSaleDurantionExtentionLimit > 0);
+        uint256 extention = extentionInDays.mul(1 days); // convert the days in minutes
+        mainSaleDurantionExtentionLimit = mainSaleDurantionExtentionLimit.sub(extentionInDays); // substract days from the limit
```

Comment:
We can have separate require if the limit - extension >= 0 . Currently safemath will throw here, because the result will be negative if we use higher extension, but will be better to have it explicitly stated in require.